ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/network/DefaultP2PNetwork.java

```diff
@@ -259,6 +259,10 @@ public void start() {
     peerBondedObserverId =
         OptionalLong.of(peerDiscoveryAgent.observePeerBondedEvents(this::handlePeerBondedEvent));
 
+    // Call checkMaintainedConnectionPeers() now that the local node is up, for immediate peer
+    // additions
+    checkMaintainedConnectionPeers();
```

Comment:
this might make it faster - have you observed this in practice ie on a running node?

Reply:
Yes, in the absence of this call, the node does not establish peering until the first scheduled run of maintained connections:

```
2022-09-30 21:43:31.530+00:00 | main | INFO  | DefaultSynchronizer | Starting synchronizer.
2022-09-30 21:43:31.531+00:00 | main | INFO  | FullSyncDownloader | Starting full sync.
2022-09-30 21:43:31.533+00:00 | main | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
2022-09-30 21:43:31.544+00:00 | main | INFO  | JsonRpcHttpService | Starting JSON-RPC service on 0.0.0.0:8555
2022-09-30 21:43:31.736+00:00 | vert.x-eventloop-thread-1 | INFO  | JsonRpcHttpService | JSON-RPC service started and listening on 0.0.0.0:8555
2022-09-30 21:43:31.737+00:00 | main | INFO  | WebSocketService | Starting Websocket service on 0.0.0.0:8556
2022-09-30 21:43:31.740+00:00 | vert.x-eventloop-thread-1 | INFO  | WebSocketService | Websocket service started and listening on 0.0.0.0:8556
2022-09-30 21:43:31.741+00:00 | main | INFO  | Runner | Ethereum main loop is up.
2022-09-30 21:43:31.793+00:00 | main | INFO  | AutoTransactionLogBloomCachingService | Starting auto transaction log bloom caching service.
2022-09-30 21:43:31.797+00:00 | main | INFO  | LogBloomCacheMetadata | Lookup cache metadata file in data directory: /qdata/ethereum/caches
2022-09-30 21:43:36.538+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
2022-09-30 21:43:41.539+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
2022-09-30 21:43:46.540+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
2022-09-30 21:43:51.541+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
2022-09-30 21:43:56.542+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
2022-09-30 21:44:01.544+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
2022-09-30 21:44:06.544+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
2022-09-30 21:44:11.546+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
2022-09-30 21:44:16.547+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
2022-09-30 21:44:21.548+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
2022-09-30 21:44:26.550+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
2022-09-30 21:44:31.551+00:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 0
2022-09-30 21:44:35.025+00:00 | nioEventLoopGroup-3-5 | INFO  | FullSyncTargetManager | No sync target, waiting for peers: 1
2022-09-30 21:44:40.415+00:00 | nioEventLoopGroup-3-5 | INFO  | SyncTargetManager | Found common ancestor with peer Peer 0xc39364ec1ed6c1d02f... at block 27011
```

Reply:
good find!